diff -Naur petsc-3.10.2/src/sys/objects/init.c petsc-3.10.2-mpifix/src/sys/objects/init.c
--- petsc-3.10.2/src/sys/objects/init.c	2018-09-12 06:14:52.000000000 -0700
+++ petsc-3.10.2-mpifix/src/sys/objects/init.c	2020-04-25 23:33:54.604409234 -0700
@@ -170,14 +170,14 @@
   in the debugger hence we call abort() instead of MPI_Abort().
 */
 
-void Petsc_MPI_AbortOnError(MPI_Comm *comm,PetscMPIInt *flag)
+void Petsc_MPI_AbortOnError(MPI_Comm *comm,PetscMPIInt *flag,...)
 {
   PetscFunctionBegin;
   (*PetscErrorPrintf)("MPI error %d\n",*flag);
   abort();
 }
 
-void Petsc_MPI_DebuggerOnError(MPI_Comm *comm,PetscMPIInt *flag)
+void Petsc_MPI_DebuggerOnError(MPI_Comm *comm,PetscMPIInt *flag,...)
 {
   PetscErrorCode ierr;
 
@@ -414,7 +414,7 @@
     MPI_Errhandler err_handler;
 
     ierr = PetscSetDebuggerFromString(string);CHKERRQ(ierr);
-    ierr = MPI_Comm_create_errhandler((MPI_Handler_function*)Petsc_MPI_DebuggerOnError,&err_handler);CHKERRQ(ierr);
+    ierr = MPI_Comm_create_errhandler(Petsc_MPI_DebuggerOnError,&err_handler);CHKERRQ(ierr);
     ierr = MPI_Comm_set_errhandler(comm,err_handler);CHKERRQ(ierr);
     ierr = PetscPushErrorHandler(PetscAttachDebuggerErrorHandler,0);CHKERRQ(ierr);
   }
@@ -464,7 +464,7 @@
       } else {
         ierr = PetscStopForDebugger();CHKERRQ(ierr);
       }
-      ierr = MPI_Comm_create_errhandler((MPI_Handler_function*)Petsc_MPI_AbortOnError,&err_handler);CHKERRQ(ierr);
+      ierr = MPI_Comm_create_errhandler(Petsc_MPI_AbortOnError,&err_handler);CHKERRQ(ierr);
       ierr = MPI_Comm_set_errhandler(comm,err_handler);CHKERRQ(ierr);
     }
     ierr = PetscFree(nodes);CHKERRQ(ierr);
